python/runtime/pai/submitter.py

```diff
@@ -377,3 +378,139 @@ def submit_pytf_train(datasource, estimator, select, validation_select,
+    """
+    # FIXME(typhoonzero): if the model not exist on OSS, assume it's a random forest model
+    # should use a general method to fetch the model and see the model type.
+    bucket = model.get_models_bucket()
+    tf = bucket.object_exists(model_name + "/tensorflow_model_desc")
+    if tf:
+        modelType = model.MODEL_TYPE_TF
+        bucket.get_object_to_file(
+            model_name + "/tensorflow_model_desc_estimator",
+            "tmp_estimator_name")
+        with open("tmp_estimator_name") as file:
+            estimator = file.readline()
+        return modelType, estimator
+
+    xgb = bucket.object_exists(model_name + "/xgboost_model_desc")
+    if xgb:
+        modelType = model.MODEL_TYPE_XGB
+        return modelType, ""
+
+    return model.MODEL_TYPE_PAIML, ""
+
+
+def get_pai_predict_cmd(datasource, select, model_name, predict_table,
+                        result_table, model_type):
+    """Get predict command for PAI task
+
+    Args:
+        datasource: current datasource
+        select: sql statement to get predict data set
+        model_name: model used to do prediction
+        predict_table: where to store the tmp prediction data set
+        result_table: prediction result
+        model_type: type of th model, see also get_oss_saved_model_type
+
+    Returns:
+        The command to submit PAI prediction task
+    """
+    # NOTE(typhoonzero): for PAI machine learning toolkit predicting, we can not load the TrainStmt
+    # since the model saving is fully done by PAI. We directly use the columns in SELECT
+    # statement for prediction, error will be reported by PAI job if the columns not match.
+    schema = db.get_table_schema(datasource, result_table)
+    result_fields = [col[0] for col in schema]
+
+    if model_type == model.MODEL_TYPE_PAIML:
+        return ('''pai -name prediction -DmodelName="%s"  '''
+                '''-DinputTableName="%s"  DoutputTableName="%s"  '''
+                '''-DfeatureColNames="%s"  -DappendColNames="%s"''') % (
+                    model_name, predict_table, result_table,
+                    ",".join(result_fields), ",".join(result_fields))
+    else:
+        # (TODO:lhw) add cmd for other model_type
+        raise SQLFlowDiagnostic("not implemented")
+
+
+def create_predict_result_table(datasource, select, result_table,
+                                label_column):
+    """Create predict result table with given name and label column
+
+    Args:
+        datasource: current datasource
+        select: sql statement to get prediction data set
+        result_table: the table name to save result
+        label_column: name of the label column, if not exist in select
+            result, we will add a int column in the result table
+    """
+    conn = db.connect_with_data_source(datasource)
+    db.exec(conn, "DROP TABLE IF EXISTS %s" % result_table)
+    create_table_sql = "CREATE TABLE %s AS SELECT * FROM %s LIMIT 0" % (
+        result_table, select)
```

Comment:
The result label name may be different from the label name used when training.

Reply:
Line 472 add a column when the result label is not present in the datasource

Reply:
We may have below cases when predicting:

1. train with `select f1, f2, label TO TRAIN ...` and predict with `select f1, f2 TO PREDICT pred.pred_result`, the label column is not selected int the predict select statement.
1. train with `select f1, f2, label TO TRAIN ...` and predict with `select f1, f2, label TO PREDICT pred.label`, label column is selected and the output column name is the same as the train label column.
1. train with `select f1, f2, label TO TRAIN ...` and predict with `select f1, f2, label TO PREDICT pred.result`, label column is selected but the output column name is different with the train label column.